additional-providers/hybridauth-openstreetmap/Providers/OpenStreetMap.php

```diff
@@ -0,0 +1,85 @@
+<?php
+/*!
+* HybridAuth
+* http://hybridauth.sourceforge.net | http://github.com/hybridauth/hybridauth
+* (c) 2017, HybridAuth authors | http://hybridauth.sourceforge.net/licenses.html 
+*/
+
+/**
+* Hybrid_Providers_OpenStreetMap (openstreetmap.org)
+*/
+class Hybrid_Providers_OpenStreetMap extends Hybrid_Provider_Model_OAuth1
+{
+   	/**
+	* IDp wrappers initializer 
+	*/
+	function initialize()
+	{
+		parent::initialize();
+
+		// provider api end-points
+		$this->api->api_base_url      = "https://api.openstreetmap.org/api/0.6/";
+		$this->api->authorize_url     = "https://www.openstreetmap.org/oauth/authorize";
+		$this->api->request_token_url = "https://www.openstreetmap.org/oauth/request_token";
+		$this->api->access_token_url  = "https://www.openstreetmap.org/oauth/access_token";
+
+		// turn off json parsing!
+		$this->api->decode_json = false;
+	}
+
+
+   /**
+	* load the user profile from the api client
+	*/
+	function getUserProfile()
+	{ 
+		try{  
+			$apiUrl = 'https://api.openstreetmap.org/api/0.6/user/details';
```

Comment:
can be just `'user/details'` because `$this->api->api_base_url` will be added automatically.

Reply:
That's true - I've add several issue when making this provider, so I tried several things. The real fix was in fact to switch to a XML decoder instead of using json (which seems to be the default decoding)